Content.Server/Polymorph/Systems/PolymorphSystem.cs

```diff
@@ -121,6 +125,24 @@ private void OnBeforeFullyEaten(EntityUid uid, PolymorphedEntityComponent comp,
             }
         }
 
+        /// <summary>
+        /// It is possible to be polymorphed into an entity that can't "die", but is instead
+        /// destroyed. This handler ensures that destruction is treated like death.
+        /// </summary>
+        private void OnDamageThresholdReached(EntityUid uid, PolymorphedEntityComponent comp, DamageThresholdReached args)
```

Comment:
wont this trigger on every threshold change even if it doesnt destroy the entity? this should subscribe to `DestructionEventArgs` instead